File: 2015/java/1/Main.java
import java.io.File;
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        // Get command line arguments
        String[] arguments = args;

        if (arguments.length < 1) {
            System.out.println("Usage: java Main <input_file>");
            return;
        }

        // open file
        String filename = arguments[0];
        File file = new File(filename);
        try {
            Scanner scanner = new Scanner(file);

            var floor = 0;
            var count = 0;
            var basement = 0;
            // read file
            while (scanner.hasNextLine()) {
                String line = scanner.nextLine();
                for (int i = 0; i < line.length(); i++) {
                    if (line.charAt(i) == '(') {
                        floor++;
                    } else if (line.charAt(i) == ')') {
                        floor--;
                    }

                    count++;
                    if (floor == -1 && basement == 0) {
                        basement = count;
                    }
                }
            }

            System.out.println("Part 1: " + floor);
            System.out.println("Part 2: " + basement);

        } catch (Exception e) {
            System.out.println("Error: " + e);
        }

    }
}

File: 2015/go/1/main.go
package main

import (
	"fmt"
	"os"
)

func main() {
	args := os.Args[1:]
	if len(args) < 1 {
		fmt.Println("Usage: go run main.go <input_file>")
		return
	}

	file, err := os.Open(args[0])
	if err != nil {
		fmt.Println(err)
		return
	}

	// loop line by line
	floor := 0
	basement := 0
	count := 0
	for {
		var line string
		_, err := fmt.Fscanln(file, &line)
		if err != nil {
			break
		}

		// loop char by char
		for _, c := range line {
			if c == '(' {
				floor++
			} else if c == ')' {
				floor--
			}

			count++
			if floor == -1 && basement == 0 {
				basement = count
			}
		}
	}

	fmt.Println("Part 1:", floor)
	fmt.Println("Part 2:", basement)
}
